R/CDISCFilteredData.R

```diff
@@ -237,18 +239,29 @@ CDISCFilteredData <- R6::R6Class( # nolint
       dataset_args[["parent"]] <- NULL
 
       #TODO what happens if parent dataset doesn't actually exist...
-      super$set_dataset(dataset_args, dataname)
       private$parents[[dataname]] <- parent_dataname
+      super$set_dataset(dataset_args, dataname)
 
-      if (length(parent_dataname) > 0) {
-        parent_dataset <- self$get_filtered_dataset(parent_dataname)
-        fdataset <- self$get_filtered_dataset(dataname)
-        fdataset$add_to_eval_env(
-          parent_dataset$get_filtered_dataname(),
-          list(parent_dataset$get_data_reactive())
-        )
-      }
+      #if (length(parent_dataname) > 0) {
+      #  parent_dataset <- self$get_filtered_dataset(parent_dataname)
+      #  fdataset <- self$get_filtered_dataset(dataname)
+      #  fdataset$add_to_eval_env(
+      #    parent_dataset$get_filtered_dataname(),
+      #    list(parent_dataset$get_data_reactive())
+      #  )
+      # }
     }
+
+  #' @description
+  #' Adds objects to the filter call evaluation environment
+  #' @param name (`character`) object name
+  #' @param value object value
+  #' @return invisibly this `FilteredDataset`
+  #add_to_eval_env = function(name, value) {
+  #  checkmate::assert_string(name)
+  #    private$eval_env <- c(private$eval_env, setNames(value, name))
+  #    invisible(self)
+  #  }
```

Comment:
This needs to be replaced so that we get the filtered dataset in the environment

![image](https://user-images.githubusercontent.com/15201933/174056426-94dfd434-3fa1-444d-8fee-c6d2cce2e0a2.png)